pkg/azmanager/azmanager.go

```diff
@@ -30,8 +34,50 @@ const (
 	LBBackendPoolIDTemplate = "/subscriptions/%s/resourceGroups/%s/providers/Microsoft.Network/loadBalancers/%s/backendAddressPools/%s"
 	// LB probe ID template
 	LBProbeIDTemplate = "/subscriptions/%s/resourceGroups/%s/providers/Microsoft.Network/loadBalancers/%s/probes/%s"
+
+	defaultPollInterval = 10 * time.Second
+	defaultPollTimeout  = 2 * time.Minute
+
+	ErrRateLimitReached = "rate limit reached"
 )
 
+func isRetriableError(err error) bool {
+	return err.Error() == ErrRateLimitReached
+}
+
+type retrySettings struct {
+	Interval *time.Duration
+	Timeout  *time.Duration
+}
+
+func wrapRetry(ctx context.Context, operationName string, operation func(context.Context) error, retrySettings ...retrySettings) error {
+	interval := defaultPollInterval
+	timeout := defaultPollTimeout
+	if len(retrySettings) > 0 {
+		if retrySettings[0].Interval != nil {
+			interval = *retrySettings[0].Interval
+		}
+		if retrySettings[0].Timeout != nil {
+			timeout = *retrySettings[0].Timeout
+		}
+	}
+	return wait.PollUntilContextTimeout(ctx, interval, timeout, true, func(ctx context.Context) (bool, error) {
```

Comment:
instead of a fixed interval, could we use a backoff retry?

Reply:
updated to use backoff by default